impl/src/main/java/io/jsonwebtoken/impl/DefaultJwtParser.java

```diff
@@ -99,7 +99,9 @@ public DefaultJwtParser() { }
                      CompressionCodecResolver compressionCodecResolver) {
         this.signingKeyResolver = signingKeyResolver;
         this.key = key;
-        this.keyBytes = keyBytes;
+        if (keyBytes != null) {
+            this.keyBytes = keyBytes.clone();
```

Comment:
Why was this added?  Could be a minor performance hit.

Reply:
to prevent `keyBytes` from mutating at some point later on. The parser is meant to be reused, so perf _shouldn't_ be an issue, but... YMMV